### PR TITLE
Adding power operator with ^  and code style standardisation 

### PR DESCRIPTION
**Pull request by znichola**
[adding power opperator](https://github.com/sophiakoulen/mathematical-expression-parser/commit/174056642e0c640c3000a0bd3293ec5a0b295a97)

### CODE REVIEW

**Comment by znichola** [comment]
Standardised the code styling a bit with the latest merge
adding () to return
only using tabs for indentation
adding newline as last line in file
only using LF for newline character changed from CRLF in some files

File: print_tree.c [hunk 1]
```diff
@@ -1,143 +1,144 @@
...
-	ret[3] = "/";
-	ret[4] = "-";
-	return (ret[index]);
-}
-
-*/
-
-static void	print_token(t_token *tok)
-{
-	// printf("%.5s\n%.5s\n%.5s\n", dict_type(tok), dict_category(tok), dict_name(tok));
```
Comment (sophiakoulen) [inline print_tree.c:61]: Please remove commented code if you won't need it

File: parser.c [hunk 2]
```diff
@@ -68,7 +68,7 @@ int	parse_term(char **str, t_tree **left_tree)
 		tok = scan_token(*str);
 		if (!tok)
 			return (0);
-		if (!(tok->type == operation && (tok->value.c == '*' || tok->value.c == '/')))
+		if (!(tok->type == operation && (tok->value.c == '*' || tok->value.c == '/' || tok->value.c == '^')))
```
Comment (sophiakoulen) [inline parser.c]: The priorities of operations aren't respected. In the case `./eval "4 * 3 ^ 2"`, 3 ^2 should have priority over multiplication. I think power operation cannot be handled in the parse_term function.

**Comment by sophiakoulen** [comment]
I think it's a great idea to add the power operation. However, that cannot be handled inside the parse_term function like you did. I believe it's best if we rethink the grammar first and then update the code.

**Review by sophiakoulen — CHANGES_REQUESTED** [review]

File: parser.c (hunk 2 above)
Comment (znichola) [reply parser.c]: yes, I think it's easier to remove the operator in this pull request and sort it out later.

**Review by znichola — COMMENTED** [review]
Removed the uneeded commented code and removed the power operator, to be added later.

File: print_tree.c (hunk 1 above)
Comment (znichola) [reply print_tree.c:61]: done!

**Comment by znichola** [comment]
Ok I think it's possible to merge now, I removed the commented code in print_tree and removed the power operator to be added later when after a re-think.

**Review by znichola — COMMENTED** [review]
Ah I had to also review this file. It's done now I think.

**Pull request merged**